app/utils/Assets.scala

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import java.net.URL
+import controllers.CachedAssets.hashedPaths
+import play.api.Play
+import org.apache.commons.io.IOUtils
+
+object Assets {
+
+  def inlineResource(path: String) = {
+    val url = AssetFinder("public/" + hashedPaths.getOrElse(path, path))
+    IOUtils.toString(url)
+  }
+
+  val svgSprite = inlineResource("images/svg-sprite.svg")
+}
+
+object AssetFinder {
+  def apply(assetPath: String): URL = {
+    Option(Play.classloader(Play.current).getResource(assetPath)).getOrElse {
+      throw AssetNotFoundException(assetPath)
+    }
+  }
+}
+
+case class AssetNotFoundException(assetPath: String) extends Exception {
```

Comment:
TIP: if you don't want to bother creating an exception class ex-novo, and you cannot find an already existing one which fits the bill, consider using Scala's `sys.error` method, which will throw a generic runtime exception